File: backend/internal/models/connection.go
package models

type AuthMethod string

const (
	AuthPassword  AuthMethod = "password"
	AuthPublicKey AuthMethod = "publickey"
)

type ConnectionConfig struct {
	ID         string     `json:"id"`
	Name       string     `json:"name"`
	Host       string     `json:"host"`
	Port       int        `json:"port"`
	Username   string     `json:"username"`
	AuthMethod AuthMethod `json:"auth_method"`
	PrivateKey string     `json:"private_key,omitempty"`
	Group      string     `json:"group,omitempty"`
	
	// Runtime-only fields (not persisted to JSON)
	Password   string `json:"-"`
	Passphrase string `json:"-"`
}

File: mobile/src/types/connection.ts
export type AuthMethod = 'password' | 'publickey'

export interface SessionProfile {
  term?: string
  font_size?: number
  startup_command?: string
  startup_command_delay_ms?: number
  terminal_theme?: string
}

export interface ConnectionConfig {
  id: string
  name: string
  host: string
  port: number
  username: string
  auth_method: AuthMethod
  private_key?: string
  key_id?: string
  password?: string
  group?: string
  profile?: SessionProfile
}
